src/python/gudhi/representations/vector_methods.py

```diff
@@ -885,3 +885,64 @@ def transform(self, X, sample_weight=None):
 
     def get_feature_names_out(self):
         return self._running_transform_names
+
+class PersistenceLengths(BaseEstimator, TransformerMixin):
+    """
+    This is a class that returns the N-longest persistence lengths.
+    """
+    def __init__(self, num_lengths=5):
+        """
+        Constructor for the PersistenceLengths class.
+
+        Parameters:
+            num_lengths (int): number of persistence lengths to return (default 5).
+        """
+        self.num_lengths  = num_lengths
+
+    def fit(self, X, y=None):
+        """
+        Fit the PersistenceLengths class on a list of persistence diagrams (this function actually does nothing but is
+        useful when PersistenceLengths is included in a scikit-learn Pipeline).
+
+        Parameters:
+            X (list of n x 2 or n x 1 numpy arrays): input persistence diagrams.
+            y (n x 1 array): persistence diagram lengths (unused).
+        """
+        return self
+
+    def transform(self, X):
+        """
+        Compute the persistence landscape for each persistence diagram individually and concatenate the results.
+
+        Parameters:
+            X (list of n x 2 numpy arrays): input persistence diagrams.
+    
+        Returns:
+            numpy array with shape (number of diagrams) x (num_lengths): output persistence lengths.
+        """
+
+        pers_lengths = []
+        for pd in X:
+            # Sort in reverse order persistence lengths (where length = death - birth)
+            lengths = np.flip(np.sort(pd[:,1] - pd[:,0]))
```

Comment:
To get the top k elements (with k significantly smaller than n), a slightly more efficient algorithm is to use partition(*, -k)[-k:] (all the numbers I give are ±1, to be checked) and then you only need to sort those k elements. But that can come later if you prefer.

Reply:
I did so on [768774a](https://github.com/GUDHI/gudhi-devel/pull/1117/commits/768774a184139fcddd168d0ee6e5e4f5ce8eaa15), but I had to assert on `num_lengths <= 0` in the constructor. To be discussed, cf. [this discussion in Ripser PR](https://github.com/GUDHI/gudhi-devel/pull/920#discussion_r1741115600)